include/libvhd.h

```diff
@@ -165,21 +163,21 @@ struct vhd_context {
 };
 
 static inline int
-test_bit (volatile char *addr, int nr)
+test_bit (const void *addr, uint32_t nr)
 {
-	return ((addr[nr >> 3] << (nr & 7)) & BIT_MASK) != 0;
+	return (((const uint8_t *)addr)[nr >> 3] >> (nr & 7)) & 1;
```

Comment:
If you can bare to swap `int` for `bool`, you can drop the final `& 1`.

Reply:
yep, I had actually intended to do just that and forgot